### DISCUSSION TITLE
Webradio random drop outs / interruptions

### DISCUSSION THREAD

**Discussion by Roland-B** [rern/rAudio#64] [Fixed Issues] [answered]
Hi Rern,

Happy new year, thanks for bringing us this awsome build and may it continue for another year at the least :)

I mentioned this previously and thought it was fixed, but on listening to Webradio now for longer periods I get random drop outs / interruptions of about a second of silence. I have gone back to previous build of Rune and the radio is fine so is not my internet conection (using LAN cable) or the station (BBC Radio 6). To me it appears as if is being limited by processor or come kind of processing somewhere else in rune interfering/using cpu. I can make it get worse when it occurs by switching tabs in a web browser (Firefox and Chrome) from rune to another tab and back and forth, back and forth quickly, so "seems" to be throttling/limiting of some kind as CPU on system page seems to be low. 

Thanks for your assistance with this as always

RolandB

rAudio 1 • 20210101
Raspberry Pi 4 Model B Rev 1.2
BCM2711 • 4 Cortex-A72 @ 1.5GHz • 4GB
E30
5.4.83-3-ARCH

**Comment by R101G** [comment]
Just a suggestion, but have you checked the CPU temperature? I had so many problems using the Pi4 that I went back to using a Pi3.

**Comment by Roland-B** [comment]
@R101-Github

Thanks for suggestion, most appreciated bud, but is fine when I use previous build and works fine for all other uses of current build inc large/ quality flac files, should be no issue for a 320Kbps stream. Also is running cooler than a polar bears toe-nails at 40.9 °C in use on average as have aluminium case with heat sinks.

RolandB

**Comment by klassikmann** [comment]
@Roland-B I made the same experience (random drop outs) on BBC Radio 3 since some weeks. I'll investigate the stream to sort out rAudio as the cause. I'll come back if I have some insights.

@R101-Github CPU temperature is moderate (51° C).

**Comment by klassikmann** [comment]
@Roland-B Could you please post the "BBC Radio 6" URL you are using?

**Reply by Roland-B** [reply]
Hi @klassikmann,

http://a.files.bbci.co.uk/media/live/manifesto/audio/simulcast/hls/uk/sbr_high/ak/bbc_6music.m3u8

I have have tried lots of different stations other than BBC and they all eventually do it, even the station that comes as standard with the build. I use the radio all day while working from home and it became so annoying I went back to re6 which works perfectly for me for now.

Thanks for you assistance in trying to find a fix.

RolandB

**Reply by ceejayemm** [reply]
Roland

I use the following urls for BBC Radio 2 and BBC 6 Music and can't say I have noticed dropouts (RPi 3B+, Merus Audio Amp)

BBC 6 Music
http://as-hls-uk-live.bbcfmt.s.llnwi.net/pool_904/live/uk/bbc_6music/bbc_6music.isml/bbc_6music-audio%3d320000.norewind.m3u8

BBC Radio 2
http://as-hls-uk-live.bbcfmt.s.llnwi.net/pool_904/live/uk/bbc_radio_two/bbc_radio_two.isml/bbc_radio_two-audio%3d320000.norewind.m3u8

Chris

**Comment by klassikmann** [comment]
@Roland-B The BBC radio live streams are no "ordinary" streams (as the "Radio Paradise" ones) but "HTTP Live Streams" (HLS), see [https://en.wikipedia.org/wiki/HTTP_Live_Streaming](url). 

What does it mean? The outmost .m3u8 (bbc_6music.m3u8) contains a link to another .m3u8 (bbc_6music-audio%3d320000.norewind.m3u8) which describes the first portions containing consecutive few seconds (here 6 sec) of "AAC LC" audio contained in a "MPEG2 Transport Stream" (.ts) which a forwarded by MPD to the ffmpeg plugin for demuxing the TS-container and playing the AAC LC. 

If ffmpeg is not capable to fetch in time a portion of a .ts then six seconds of audio are missing. 

I can't comment to your observation that re6 is "stutter-free" because I switched to rAudio (but perhaps I burn a spare sdcard with the latest re6 for testing).

@rern Are you aware of any change in the relationship mpd <-> ffmpeg during the last weeks?

EDIT: Correct "MPEG1 Transport Stream" to "MPEG2 Transport Stream"

**Reply by Roland-B** [reply]
@klassikmann 


That all sounds plausable, and great work Thanks! But I have found it happens with all radio stations.. I gave those Radio Stations as just a few examples.

http://mp3channels.webradio.antenne.de/chillout

This as far as I can tell is just a standard mp3 stream, and also does it.

I have also found it can be made worse by going to System Settings page > and clicking on Status refresh (and leave it running) and even worse when you also continuously click between another browser tab and Re1 quickly back and forth back and forth ad infinitum. So is a issue IMO with being limited/throttled somehow.. beyond that I have no ideas unfortunately. 

I can also confirm again I have listened all day on previous version (e6 20201221) with no interruptions at all, and then loaded new version rAudio-1 on another microsd card and updated to latest version and still has the issue within minutes.

Thanks again for your efforts.

RolandB

rAudio 1 • 202101021
Raspberry Pi 4 Model B Rev 1.2
BCM2711 • 4 Cortex-A72 @ 1.5GHz • 4GB
E30
5.4.83-5-ARCH

**Reply by R101G** [reply]
@Roland-B 

The Pi4 has known problems with RF interference between USB3, Bluetooth and wifi but, as it works with e6, there must be a feature enabled by rAudio which is causing a problem. That is, if you are sure the rAudio OS is not just drawing more power than e6 does.

As we are grasping at straws, and the Pi3 ethernet bottleneck problem does not exist in the Pi4, I would suggest disabling Bluetooth and wifi in /boot/config.txt (add the lines _dtoverlay=disable-wifi_ and _dtoverlay=disable-bt_) just to make sure they are not active.

Also, check that the Pi is being grounded in some way. If the DAC is properly isolated, the USB side will be isolated from the audio outputs, so it might not be.

**Reply by Roland-B** [reply]
@R101-Github 

Hi, Thanks for your suggestions but I'm 99% sure is a build issue, but i'm willing to grasp a staw or 2 :) Will also point out that this issue does not happen on Pi4 using other music players whose names I wont mention here ;)

Wifi and Bluetooth are already disabled on both builds, but I will give the boot/config.txt change a go tommorow and report back IF makes a difference which I doubt tbh.

Even if newer build does somehow use more power this should not impact performance if an adiquate power supply and cable is used, which I believe I have.  My DAC has its own power supply and is using the USB2 port for data, and nothing else plugged in. If you know of a way to test this though both I would be happy give it a go.

Grounding shoudn't be an issue as it would not resolve itself consistantly with a change with the version of rune build being used?

Thanks,

RolandB

**Reply by R101G** [reply]
@Roland-B 

This can't just be a software thing, or other rAudio users would have the same problem. It seems that something enabled by rAudio, but by none of the other builds, is affecting your particular hardware. Having experienced problems due to the Pi4 generating RF interference, I would also suggest trying it without the aluminium box!

**Reply by Roland-B** [reply]
@R101-Github

We will have to agree to disagree then, and I wait for a response from @rern but thanks for your input.

RolandB

**Reply by rern** [reply]
Try update kernel and packages:
`pacman -Syu`

**Reply by Roland-B** [reply]
@rern 

Thanks for reply. Have done this morning, and applied latest update and rebooted and the problem still exists. 

Attached recording of the issue.

https://user-images.githubusercontent.com/43496391/103998426-e8a37180-5193-11eb-9ba3-0b9b63785bdd.mp4

RolandB

rAudio 1 • 20210107
Raspberry Pi 4 Model B Rev 1.2
BCM2711
4 Cortex-A72 @ 1.5GHz • 4GB
5.4.83-5-ARCH

**Reply by klassikmann** [reply]
> 
> 
> Try update kernel and packages:
> `pacman -Syu`

@rern What did you expect of the full update or hope for? mpd and ffmpeg are still the same.
I did it and the drop-outs are still there with the BBC HLS 320kb AAC LC streams (other webradios were never affected here). If I store such a stream using ffmpeg, I have to face the same effects.
btw.: The specification of HLS can be found in RFC 8216.

Possible causes:
- These streams are geo-blocked for UK, so I use a VPN for access. Perhaps the connection isn't performant enough. 
- ffmpeg has problems with streams with a low EXT-X-TARGETDURATION (here 6sec). I found an 4 year old issue (https://trac.ffmpeg.org/ticket/5932) which might correlate.

**Reply by Roland-B** [reply]
@klassikmann @rern 

I am in UK and have a good connection.. hope this helps narrow down the issue.

RolandB

**Reply by R101G** [reply]
Could someone clarify whether this affects all stations, or just the BBC ones? I have a radio stream going for most of the day and don't experience the problem, but I rarely listen to the BBC streams.

**Reply by ceejayemm** [reply]
I am in the UK and have 2 systems based on RPi 3B+, one wireless with Merus Audio Amp and the other wired with IQAudIO DAC+ and listen to BBC Radio 2 most days on one or the other. As noted above I use a slightly different url to you but haven't noticed any dropouts so far since I moved to RAudio-1 (from RA+Re6). Both are in plastic cases.

Chris

**Reply by findnix** [reply]
I can confirm the same behavior. 
I also observe dropouts when playing from NAS via SMB (1 Gbit/s) or internal m.2 SSD (>250 MByte/s in an argonONE case). I first suspected HiRes FLAC and DSD. However, dropouts also occur with 128k mp3. Temperature is around 50°C.
I also tested via HDMI and cross swapped all cables to rule out infrastructure errors. To exclude effects on the USB bus I tested without internal disk. Without success.
On RPi3 and RPi2 with e3 and the same DAC and the same sources everything runs to full satisfaction.
The journal is full with "taskset[439]: alsa_output: Decoder is too slow; playing silence to avoid xrun".

Greetings from Austria

**Reply by rern** [reply]
@klassikmann,

Multiple updates in a row recently:
- linux-raspberrypi4
- linux-raspberrypi4-headers
- raspberrypi-bootloader
- raspberrypi-bootloader-x
- raspberrypi-firmware

Why did you think it's FFMpeg? @Roland-B runs R e6 with no dropouts.

**Reply by findnix** [reply]
Updated my PI4 to latest firmware with parallel update to 20210108 ... testing now ... without success.
Clear dropouts after a couple of minutes in a DSD64 file from SSD and SMB.
About 25% CPU usage of "mpd", all other CPUs almost idle. 
Same file plays fine on e3/RPi3

```
rAudio:~  #  rpi-eeprom-update
BCM2711 detected
VL805 firmware in bootloader EEPROM
BOOTLOADER: up-to-date
CURRENT: Thu Apr 16 17:11:26 UTC 2020 (1587057086)
 LATEST: Thu Apr 16 17:11:26 UTC 2020 (1587057086)
 FW DIR: /lib/firmware/raspberrypi/bootloader/critical
VL805: up-to-date
CURRENT: 000137ad
 LATEST: 000137ad
```

**Reply by Roland-B** [reply]
> 
> 
> Could someone clarify whether this affects all stations, or just the BBC ones? I have a radio stream going for most of the day and don't experience the problem, but I rarely listen to the BBC streams.

@R101-Github 

My post on this thread 2 days ago.

"That all sounds plausable, and great work Thanks! But I have found it happens with all radio stations.. I gave those Radio Stations as just a few examples.

http://mp3channels.webradio.antenne.de/chillout

This as far as I can tell is just a standard mp3 stream, and also does it."

RolandB

**Reply by Roland-B** [reply]
> 
> 
> I can confirm the same behavior.
> I also observe dropouts when playing from NAS via SMB (1 Gbit/s) or internal m.2 SSD (>250 MByte/s in an argonONE case). I first suspected HiRes FLAC and DSD. However, dropouts also occur with 128k mp3. Temperature is around 50°C.
> I also tested via HDMI and cross swapped all cables to rule out infrastructure errors. To exclude effects on the USB bus I tested without internal disk. Without success.
> On RPi3 and RPi2 with e3 and the same DAC and the same sources everything runs to full satisfaction.
> The journal is full with "taskset[439]: alsa_output: Decoder is too slow; playing silence to avoid xrun".
> 
> Greetings from Austria

Hi @findnix,

Thanks for that info, using the the detail you provided I have figured out how to check the logs and have the same, hopefully this info helps @rern figure the cause.

Jan 08 09:26:06 rAudio taskset[430]: ffmpeg/hls: skipping 1 segments ahead, expired from playlists
Jan 08 09:26:08 rAudio taskset[430]: alsa_output: Decoder is too slow; playing silence to avoid xrun
Jan 08 09:26:35 rAudio taskset[430]: alsa_output: Decoder is too slow; playing silence to avoid xrun
Jan 08 09:27:02 rAudio taskset[430]: alsa_output: Decoder is too slow; playing silence to avoid xrun
Jan 08 09:27:32 rAudio taskset[430]: alsa_output: Decoder is too slow; playing silence to avoid xrun
Jan 08 09:27:46 rAudio taskset[430]: alsa_output: Decoder is too slow; playing silence to avoid xrun
Jan 08 09:27:47 rAudio taskset[430]: ffmpeg/hls: skipping 1 segments ahead, expired from playlists
Jan 08 09:28:13 rAudio taskset[430]: alsa_output: Decoder is too slow; playing silence to avoid xrun
Jan 08 09:28:18 rAudio taskset[430]: alsa_output: Decoder is too slow; playing silence to avoid xrun
Jan 08 09:29:14 rAudio taskset[430]: alsa_output: Decoder is too slow; playing silence to avoid xrun
Jan 08 09:29:20 rAudio taskset[430]: alsa_output: Decoder is too slow; playing silence to avoid xrun
Jan 08 09:29:28 rAudio taskset[430]: alsa_output: Decoder is too slow; playing silence to avoid xrun

Edit: Also have just tested a brand new microsd card just in case was issues with slow read/write causing issue. No change.

RolandB.

**Reply by ceejayemm** [reply]
I am not seeing any of these ' Decoder too slow' errors on my installations. Both my DACs are HATs and are attached directly to the GPIO pins of my RPIs. Could it be that these errors are from a USB based DAC ?

Chris

**Reply by Roland-B** [reply]
@ceejayemm 

I wouldn't have thought so as all works perfect on R e6. The only difference for me is the new build of rune (rAudio1), and the only other factor/difference would have been the alternate microsd card at fault which I have just tested.

RolandB

**Reply by ceejayemm** [reply]
This might not be a RAudio problem.  There are similarly reported issues in the Moode forums, see:

http://moodeaudio.org/forum/archive/index.php?thread-518.html

And in the MPD Github Issues:

https://github.com/MusicPlayerDaemon/MPD/issues/206

Chris

**Reply by Roland-B** [reply]
All issues seem to be reported around 2018. R e6 is from late last year and works fine and it was only updated to RAudio-1 not very long ago. From what I can tell the issue seems to be for the build of rAudio-1 for the Pi4 from reading through here, and hopefully @rern should be able to work out what has changed between his builds for Pi4 between r e6 and rAudio-1, whether it be code or a software/kernel update and fix/revert the change.

RolandB

**Comment by klassikmann** [comment]
Obviously we are discussing different issues.

Using my configuration - RPI 3B + HifiBerry Digi+ Pro (not a DAC but a digital-out) - I'm suffering drop-outs at the BBC Radio HLS 320kb AAC LC streams only. No problems with "classical" streams, even with the streams generated by my SAT-TV embedded in .ts (MPEG 2 Transport  Stream). btw. ffmpeg needs about 7 seconds to analyze and demux the streams before playing starts.

For BBC Radio 3 HD there are two different sources available both provided by the ffmpeg plugin of mpd. One is the above mentioned HLS, the other one is a "classical" AAC stream sharing the same content (320kb AAC LC) as the HLS and playing smoothly. 
Another difference: The AAC stream starts immediately, the HLS needs between 10 to sometimes 40 seconds before playing starts. 

Same effects for both stream types apply if I use ffmpeg directly, i.e. outside of rAudio, on ArchLinux and on Ubuntu 20.10. Therefore I believe that the cause of my problem lies outside rAudio or re6 where I detected the same issues.

**Reply by Roland-B** [reply]
@klassikmann 

Yup, appears so as the fix for Pi4 in todays update has done the trick for my issue. I hope you manage to figure out the cause of your issue, as know how annoying the interruptions can be.

RolandB

**Reply by klassikmann** [reply]
@Roland-B My problem is still there. But I'm currently working through the ffmpeg logs.

**Comment by Roland-B** [comment, marked as answer]
@rern 

Currently testing todays (20210109) update with "Fix - Audio drop-outs on Raspberry Pi 4", sounding good so far, with no interruptions :)

Only listened  for around 10 mins so far and usually encountered the issue before this, but will listen for an extended period to confirm.

Thanks.

RolandB

**Reply by findnix** [reply]
Thank you @rern 
I can confirm that. Still sounds good now after an hour.
I'm still testing, but I don't expect any dropouts.

Thanks again
Ronald

**Reply by Roland-B** [reply]
@rern

After extended listen no interruptions and no errors in journal logs. Thanks very much as always.

RolandB
